emcc.py

```diff
@@ -1600,7 +1600,7 @@ def compile_source_file(i, input_file):
           for tempf, inputf in zip(temp_files, input_files):
             safe_move(tempf[1], unsuffixed_basename(inputf[1]) + final_ending)
         else:
-          if len(input_files) == 1:
+          if len(input_files) == 1 and not shared.Building.is_ar(input_files[0][1]):
             input_file = input_files[0][1]
```

Comment:
the new stuff checked here could be checked after we define `input_file` on the next line, and use that, to avoid writing `[0][1]`.

a comment explaining the behavior would also be good here

Reply:
@kripken Done. However, unless the single input case is common, IMHO the most predictable option might be just to remove the shortcut entirely, especially since the Makefile dependency fixing code is sort of fragile (replacing instead of parsing), and it allows for more clear code.